dependencies/common/install-packages

```diff
@@ -92,25 +100,31 @@ mv $PACKAGE_ARCHIVE $PACKAGE_ARCHIVE_SHA1
 # install rmarkdown main rstudio
 # install rsconnect main rstudio
 # install renv master rstudio
+cat <<- EOF | R -s
 
-# ascertain the site library. this is somewhat tricky since this script must
-# run on a variety of Linux distributions early in the Docker build process.
-# the lookup order is as follows:
-# 1. the first .Library.site, if defined; otherwise
-# 2. the last element of .libPaths()
-RPKGLIB=$(Rscript -e 'cat(if(length(.Library.site) > 0) { .Library.site[[1]] } else { tail(.libPaths(), n = 1) })')
-if [[ ! -w $RPKGLIB ]]; then
-  echo "$RPKGLIB is not writable. Creating user R library directory."
-  RPKGLIB=$(Rscript -e 'cat(path.expand(Sys.getenv("R_LIBS_USER")))')
-  mkdir -p "$RPKGLIB"
-fi
+# make sure user library exists
+dir.create(Sys.getenv("R_LIBS_USER"), recursive = TRUE, showWarnings = FALSE)
 
-# install packages needed to run tests
-for RPKG in purrr rmarkdown renv testthat xml2 yaml; do
-    RINSTALLCMD="if("'!'"require($RPKG, quietly = TRUE)) { install.packages('$RPKG', lib='$RPKGLIB', repos='https://cran.rstudio.com/') }"
-    echo "> $RINSTALLCMD"
-    Rscript -e "$RINSTALLCMD"
-done
+# set up repositories
+options(repos = c(CRAN = "https://cran.rstudio.com"))
+
+# install renv
+installed <- tryCatch(packageVersion("renv") >= "0.17.3", error = function(e) FALSE)
+if (!installed) {
+  install.packages("renv")
+}
+
+lockfile <- "${lockfile:=}"
+if (nzchar(lockfile)) {
+  # if we were provided a lockfile, use it
+  writeLines(paste("Using lockfile:", lockfile))
+  renv::restore(lockfile = lockfile)
+} else {
+  # otherwise, install preset list of packages from CRAN
+  renv::install("devtools", "rmarkdown", "renv", "testthat", "xml2", "yaml")
```

Comment:
Is this too aggressive? Maybe we should just use a regular `install.packages` call in this case, as we did before?

Reply:
(The nice thing about using `renv` is we would get to re-use the `renv` global package cache, and it would also ensure we're using the latest versions of packages locally)

Reply:
I think this is ok for builds and the cache is a plus for dev environments!